src/main/java/mods/eln/sixnode/EmergencyLamp.kt

```diff
@@ -0,0 +1,320 @@
+package mods.eln.sixnode
+
+import mods.eln.cable.CableRenderDescriptor
+import mods.eln.gui.*
+import mods.eln.i18n.I18N
+import mods.eln.i18n.I18N.tr
+import mods.eln.misc.*
+import mods.eln.node.Node
+import mods.eln.node.NodePeriodicPublishProcess
+import mods.eln.node.published
+import mods.eln.node.six.*
+import mods.eln.sim.IProcess
+import mods.eln.sim.mna.component.ResistorSwitch
+import mods.eln.sim.nbt.NbtElectricalLoad
+import mods.eln.sim.process.destruct.VoltageStateWatchDog
+import mods.eln.sim.process.destruct.WorldExplosion
+import mods.eln.sixnode.electricalcable.ElectricalCableDescriptor
+import mods.eln.sixnode.lampsupply.LampSupplyElement
+import net.minecraft.client.gui.GuiButton
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NBTTagCompound
+import org.lwjgl.opengl.GL11
+import java.io.DataInputStream
+import java.io.DataOutputStream
+
+class EmergencyLampDescriptor(name: String, val cable: ElectricalCableDescriptor, val batteryCapacity: Double,
+                              val chargePower: Double, val consumption: Double, val lightLevel: Int, model: Obj3D)
+    : SixNodeDescriptor(name, EmergencyLampElement::class.java, EmergencyLampRender::class.java) {
+
+    val mainCeiling: Obj3D.Obj3DPart = model.getPart("coreCeil")
+    val panelCeiling: Obj3D.Obj3DPart = model.getPart("panelCeil")
+    val lightCeiling: Obj3D.Obj3DPart = model.getPart("lightCeil")
+    val mainWall: Obj3D.Obj3DPart = model.getPart("coreWall")
+    val mainWallR: Obj3D.Obj3DPart = model.getPart("coreWallR")
+    val mainWallL: Obj3D.Obj3DPart = model.getPart("coreWallL")
+    val lightWall: Obj3D.Obj3DPart = model.getPart("lightWall")
+
+    init {
+        voltageLevelColor = VoltageLevelColor.fromCable(cable)
+        changeDefaultIcon("emergencylamp")
+    }
+
+    fun draw(onCeiling: Boolean = false, on: Boolean = false, mirrorSign: Boolean = false) {
+        if (onCeiling) {
+            mainCeiling.draw()
+
+            if (on)
```

Comment:
Please use braces any time the if clause is more than a single line. Or just always; I like to follow https://google.github.io/styleguide/javaguide.html#s4.1-braces on this.